docs/design/autoware-interfaces/ad-api/types/autoware_ad_api_msgs/msg/motion_factor.md

```diff
@@ -0,0 +1,37 @@
+# autoware_ad_api_msgs/msg/MotionFactor
+
+## Definition
+
+```txt
+# constants for reason
+uint16 INTERSECTION=1
+uint16 MERGE_FROM_PRIVATE_ROAD=2
+uint16 CROSSWALK=3
+uint16 WALKWAY=4
+uint16 STOP_LINE=5
+uint16 DETECTION_AREA=6
+uint16 NO_STOPPING_AREA=7
+uint16 TRAFFIC_LIGHT=8
+uint16 OBSTACLE_STOP=9
+uint16 SURROUND_OBSTACLE_CHECK=10
+uint16 BLIND_SPOT=11
+uint16 BLOCKED_BY_OBSTACLE=12
```

Comment:
`OBSTACLE_STOP` is obstacle ahead and is renamed to `FRONT_OBSTACLE`.
`BLOCKED_BY_OBSTACLE` is waiting for obstacle avoidance approval and is renamed to `AVOIDANCE1`.

Reply:
What's `1` in `AVOIDANCE1`?

Reply:
It’s temporary name and under consideration.
- `AVOIDANCE1`: A lane change to avoid obstacles when avoiding.
- `AVOIDANCE2`: A lane change to avoid obstacles when returning.

Reply:
How about these?
- `AVOIDANCE_LANE_CHANGE`
- `AVOIDANCE_LANE_RETURN`

Reply:
Since it doesn't necessarily mean LANE_CHANGE, I feel `AVOIDANCE_START(BEGIN?)/RETURN(END?)` is better.

Reply:
If that is the case, then the description should also be updated, since currently it says that this is avoidance with a lane change.

Reply:
Ah, I see. I forgot that there were two types of avoidance.
So you are correct, I agree with `AVOIDANCE_LANE_CHANGE/RETURN`! I'm sorry. :bow:

Reply:
> Since AVOIDANCE doesn't necessarily mean LANE_CHANGE,

@kenji-miyake @LalithVipulananthan 
Thank you for the comments. I forgot this case. I'll check it meets the requirements.

Reply:
There are both cases. So change it as follows.
```
AVOIDANCE_LANE_CHANGE  # with turn indicators
AVOIDANCE_LANE_RETURN  # with turn indicators
AVOIDANCE_PATH_CHANGE  # without turn indicators
AVOIDANCE_PATH_RETURN  # without turn indicators
```

Reply:
Is it possible for us to exactly judge the timing of `PATH_CHANGE/PATH_RETURN`?
I guess it's difficult because the border of the state change is ambiguous.

Reply:
I have decided not to distinguish between them now. Add the angle, width, or distance to line later if needed.